sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -888,7 +888,20 @@ public void Dispose()
                 }
 
                 target.SafeClose();
-                throw;
+
+                // The AMQP library may throw an InvalidOperationException or one of its derived types, such as
+                // ObjectDisposedException if the underlying network state changes.  While normally terminal, in this
+                // context, these exception types are safe to retry.  Translate them so that the retry policy
+                // can correctly interpret.
+
+                switch (ex)
+                {
+                    case InvalidOperationException:
```

Comment:
Is it worth patching up the IsClosed behavior that we were discussing this morning as part of this PR (so as not to make that bug worse)?

Reply:
I'm going to do those in dedicated PRs.   I suspect it'll be easier to review if we don't mash them together; I think we'll need a different approach for Event Hubs and Service Bus due to the client hierarchies.